### DISCUSSION TITLE
How to blend an image with a color or another image?

### DISCUSSION THREAD

**Discussion by seeReadCode** [marp-team/marp#268] [Ideas] [answered]
I imagine it could work like:

```
![bg](rgb(255,128,0), https://fakeimg.pl/800x600/67b8e3/fff/?text=C)
```

c.f. 
- https://css-tricks.com/basics-css-blend-modes/
- https://docs.decksetapp.com/English.lproj/Media/01-background-images.html#force-apply-filter

There might be a way to do this already – might be a feature request...

**Comment by yhatt** [comment]
Marp does not want to add a syntax that will break compatibility with Markdown for documents. We prioritize compatibility with Markdown syntax over convenience features.

Markdown image syntax `![]()` is designed for a single image (`<img src="">`) so multiple images for the image syntax you've suggested will break image rendering in documentation.

Create [Marp(it) plugin](https://marpit-api.marp.app/) for adding the suggested image syntax if you want.

### Workaround

Apply `opacity` to `<figure>`.

```markdown
![bg](rgb(255,128,0))
![bg](https://fakeimg.pl/800x600/67b8e3/fff/?text=C)

<style scoped>
figure {
  opacity: 0.5 !important;
}
</style>
```

<details>
<summary>Alternative</summary>

Apply `-webkit-mask-image` with `linear-gradient` that is using same 2 colors. Marp Core has enabled [advanced backgrounds](https://marpit.marp.app/image-syntax?id=advanced-backgrounds) so you have to apply to `<figure>` instead of `<section>`.

```markdown
![bg](rgb(255,128,0))
![bg](https://fakeimg.pl/800x600/67b8e3/fff/?text=C)

<style scoped>
figure {
  -webkit-mask-image: linear-gradient(to top, rgba(0,0,0,0.5), rgba(0,0,0,0.5)) !important;
}
</style>
```

</details>

**Comment by yhatt** [comment, marked as answer]
Oops, I have forgot about [CSS filter](https://marpit.marp.app/image-syntax?id=image-filters) `opacity`. This is the best solution:

```markdown
![bg](rgb(255,128,0))
![bg opacity](https://fakeimg.pl/800x600/67b8e3/fff/?text=C)
```
